bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go

```diff
@@ -112,8 +135,58 @@ func (c *KubeadmConfigSpec) validate(name string) error {
 		knownPaths[file.Path] = struct{}{}
 	}
 
-	if len(allErrs) == 0 {
-		return nil
+	return allErrs
+}
+
+func (c *KubeadmConfigSpec) validateIgnition() field.ErrorList {
+	var allErrs field.ErrorList
+	errIgnitionBootstrapDataNotSet := fmt.Errorf("can be set only if the IgnitionBootstrapData feature gate is enabled")
+
+	if !feature.Gates.Enabled(feature.IgnitionBootstrapData) && c.Format == Ignition {
+		allErrs = append(allErrs, field.Forbidden(
+			field.NewPath("spec", "format"), errIgnitionBootstrapDataNotSet.Error()))
+		return allErrs
 	}
-	return apierrors.NewInvalid(GroupVersion.WithKind("KubeadmConfig").GroupKind(), name, allErrs)
+
+	if !feature.Gates.Enabled(feature.IgnitionBootstrapData) && c.Ignition != nil {
+		allErrs = append(allErrs, field.Forbidden(
+			field.NewPath("spec", "ignition"), errIgnitionBootstrapDataNotSet.Error()))
+		return allErrs
+	}
```

Comment:
Can we have these two in a single if condition and declare the error in there?

Reply:
done